sycl/doc/extensions/experimental/sycl_ext_oneapi_matrix/sycl_ext_oneapi_matrix.asciidoc

```diff
@@ -0,0 +1,647 @@
+
+- General query: the general query interface provides information  about sizes, types, static/dynamic, and scopes that are supported by a specific TPU implementation. This is needed to avoid padding by the user, for tuning, and efficient code generation if used by a library. The general query return an array of `combinations` of `combination` type. Each combination includes the sizes and the types for the matrices A, B, and C. Note that for each TPU, the query returns `max_msize, max_nsize, max_ksize` or `msize, nsize, ksize` exclusively depending whether the implementation supports a continuous or discrete number of sizes. For example, Intel AMX implementation supports a continuous number of sizes so the `max_*` variant is applied and only the maximum number is returned. DPAS implementation, on the other hand, supports a discrete list of numbers so the  `msize, nsize, ksize` variant is applied.  This form takes place when users only specify the TPU they are interested in using.
+
+The table below provides a description for each of the member variables and type aliases in `tpu_params` class and the forms in which  they are defined.
+
+[frame="none",options="header"]
+|======================
+| Member/type alias in `tpu_params` | Forms they are defined in |Description
+|`type_a`| validation, default values|type alias for the type of matrix A
+|`type_b`|  validation, default values|type alias for the type of matrix B
+|`type_c`|  validation, default values|type alias for the type of matrix C
+|`defaultM`|  validation, default values|when no sizes are provided by the user, indicates the suggested default size for M; usually this corresponds to the maximum size the implementation supports. In validation mode, where the user does provide sizes, this is the same value M that the user provides if M is supported by the implementation
+|`defaultN`|  validation, default values|when no sizes are provided by the user, indicates the suggested default size for N; usually this corresponds to the maximum size the implementation supports. In validation mode, where the user does provide sizes, this is the same value N that the user provides if N is supported by the implementation
+|`defaultK`|  validation, default values|when no sizes are provided by the user, indicates the suggested default size for K; usually this corresponds to the maximum size the implementation supports. In validation mode, where the user does provide sizes, this is the same value K that the user provides if K is supported by the implementation
+|`joint_matrix_a`|  validation, default values|type alias for `joint_matrix` for matrix A
+|`joint_matrix_b`| validation, default values| type alias for `joint_matrix` for matrix B
+|`joint_matrix_c`|  validation, default values| type alias for `joint_matrix` for matrix C
```

Comment:
Should this refer to `joint_matrix_accumulator`?

Reply:
This is a good question. What do you think @gmlueck ?

Reply:
You have this inconsistency throughout the spec.  In many places, you refer to the three matrices as `a`, `b`, and `c`.  However, the `use` enum refers to them as `a`, `b`, and `accumulator`.  You should decide on one or the other and then use it consistently throughout.  Since you call the first two matrices `a` and `b`, it seems natural to me to call the third matrix `c`.  Is there a good reason to call the third matrix `accumulator`?